src/features/CourtCaseList/CourtCaseListEntry.tsx

```diff
@@ -0,0 +1,184 @@
+        <Table.Cell>
+          {triggerLockedByUsername && canUnlockCase(triggerLockedByUsername) ? (
+            <LockedByTag
+              lockedBy={triggerLockedByUsername}
+              unlockPath={unlockCaseWithReasonPath("Trigger", `${errorId}`)}
+            />
+          ) : (
+            <LockedByTag lockedBy={triggerLockedByUsername} />
+          )}
+        </Table.Cell>
+      </Table.Row>
+    ) : (
+      <></>
+    )
+
+  return (
+    <>
+      <Table.Row className={classes.caseDetailsRow}>
+        <Table.Cell>
+          <ConditionalRender isRendered={!!errorLockedByUsername}>
+            <Image src={"/bichard/assets/images/lock.svg"} width={20} height={20} alt="Lock icon" />
+          </ConditionalRender>
+        </Table.Cell>
+        <Table.Cell>
+          <Link href={caseDetailsPath(errorId)} id={`Case details for ${defendantName}`}>
+            {defendantName}
+            <br />
+            <ResolvedTag isResolved={resolutionTimestamp !== null} />
+          </Link>
+        </Table.Cell>
+        <Table.Cell>
+          <DateTime date={courtDate} dateFormat={displayedDateFormat} />
+        </Table.Cell>
+        <Table.Cell>{courtName}</Table.Cell>
+        <Table.Cell>{ptiurn}</Table.Cell>
+        <Table.Cell>
+          <UrgentTag isUrgent={isUrgent} />
+        </Table.Cell>
+        <Table.Cell>
+          <NotePreviewButton previewState={showPreview} setShowPreview={setShowPreview} numberOfNotes={numberOfNotes} />
+        </Table.Cell>
+        <Table.Cell>
+          {Object.keys(exceptions).map((code, codeId) => (
+            <GridRow key={`exception_${codeId}`}>
+              {code}
+              <b>&nbsp;{exceptions[code] > 1 ? `(${exceptions[code]})` : ""}</b>
+            </GridRow>
+          ))}
+        </Table.Cell>
+        <Table.Cell>
+          {errorLockedByUsername && canUnlockCase(errorLockedByUsername) ? (
+            <LockedByTag
+              lockedBy={errorLockedByUsername}
+              unlockPath={unlockCaseWithReasonPath("Exception", `${errorId}`)}
+            />
+          ) : (
+            <LockedByTag lockedBy={errorLockedByUsername} />
+          )}
+        </Table.Cell>
+      </Table.Row>
+      {numberOfNotes != 0 && !!showPreview && (
+        <Table.Row className={classes.notesRow}>
+          <Table.Cell style={{ paddingTop: "0px" }}></Table.Cell>
+          <Table.Cell style={{ paddingTop: "0px" }}></Table.Cell>
+          <Table.Cell style={{ paddingTop: "0px" }}></Table.Cell>
+          <Table.Cell style={{ paddingTop: "0px" }}></Table.Cell>
+          <Table.Cell style={{ paddingTop: "0px" }}></Table.Cell>
+          <Table.Cell style={{ paddingTop: "0px" }}></Table.Cell>
+          <Table.Cell style={{ paddingTop: "0px" }} colSpan={2}>
+            <NotePreview
+              latestNote={first100CharsOfMostRecentNote(userNotes)}
+              displayDate={validatedMostRecentNoteDate(mostRecentUserNote)}
+              numberOfNotes={numberOfNotes}
```

Comment:
If I understand correctly, `latestNote`,  `displayDate` and `numberOfNotes` all generated from user notes, so I wonder if we could pass userNotes to Note Preview and let it handle all the transformation?